.github/workflows/ci.yml

```
@@ -18,18 +23,60 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v2
         with:
-          python-version: 2.7
+          python-version: 3.8
 
       - name: Install Python dependencies
         run: |
           pip install --upgrade pip
-          pip install yadage
+          pip install virtualenv
+          pip install --pre reana-client
 
-      - name: Validate workflow
-        run: yadage-validate workflow/yadage/workflow.yaml | grep -q 'workflow validates'
+      - name: Validate workflows
+        run: |        
+          for file in $FILES; do reana-client validate --environments -f $file; done
+        env:
+          FILES: reana.yaml reana-slurmcern.yaml reana-htcondorcern.yaml reana-yadage.yaml reana-yadage-slurmcern.yaml reana-yadage-htcondorcern.yaml reana-cwl.yaml reana-cwl-slurmcern.yaml reana-cwl-htcondorcern.yaml reana-snakemake.yaml
 
-      - run: rm -rf results && mkdir results
-
-      - run: docker run -i --rm -v `pwd`/code:/code -v `pwd`/data:/data -v `pwd`/results:/results python:2.7-slim python /code/helloworld.py --sleeptime 0
+      - name: Run serial workflows
+        run: |
+          rm -rf results && mkdir results
+          docker run -i --rm -v `pwd`/code:/code -v `pwd`/data:/data -v `pwd`/results:/results python:2.7-slim python /code/helloworld.py --sleeptime 0
+          grep -q 'Hello Joe Bloggs!' results/greetings.txt
+          rm -rf results
+      - name: Run yadage workflows
+        run: |
+          rm -rf yadage-local-run/yadage-inputs && mkdir -p yadage-local-run/yadage-inputs
+          cd yadage-local-run
+          virtualenv ~/.virtualenvs/yadage
+          source ~/.virtualenvs/yadage/bin/activate
+          pip install yadage
+          cp -a ../code ../data yadage-inputs
+          yadage-run . ../workflow/yadage/workflow.yaml -p sleeptime=0 -p inputfile=data/names.txt -p helloworld=code/helloworld.py -d initdir=`pwd`/yadage-inputs
+          grep -q 'Hello Joe Bloggs!' helloworld/greetings.txt
+          cd ..
+          rm -rf yadage-local-run
```

You can also remove the virtualenv
```suggestion
          rm -rf yadage-local-run ~/.virtualenvs/yadage
```

Same for the other workflow languages.